frontend/src/pages/settings/users/_userId/index.vue

```diff
@@ -0,0 +1,532 @@
+                      @click="submitPassword">
+                      {{ t('save') }}
+                    </VBtn>
+                  </VCol>
+                </VRow>
+              </VContainer>
+            </VForm>
+          </VWindowItem>
+        </VWindow>
+      </VCard>
+      <VDialog
+        v-model="addTagDialogOpen"
+        width="auto">
+        <VCol class="pa-0 add-key-dialog">
+          <VCard>
+            <VCardTitle>{{ t('settings.users.tabs.parentalControl.addBlockedTag') }}</VCardTitle>
+            <VCardActions>
+              <VForm
+                class="add-key-form"
+                @submit.prevent="initializedUser.BlockedTags.push(newTagValue); addTagDialogOpen = false; newTagValue = '';">
+                <VTextField
+                  v-model="newTagValue"
+                  variant="outlined"
+                  :label="t('settings.users.tabs.parentalControl.tagName')" />
+                <VBtn
+                  color="primary"
+                  :loading="loading"
+                  :disabled="newTagValue === ''"
+                  @click="initializedUser.BlockedTags.push(newTagValue); addTagDialogOpen = false; newTagValue = '';">
+                  {{ $t('confirm') }}
+                </VBtn>
+                <VBtn @click="() => {addTagDialogOpen = false}">
+                  {{ $t('cancel') }}
+                </VBtn>
+              </VForm>
+            </VCardActions>
+          </VCard>
+        </VCol>
+      </VDialog>
+    </template>
+  </SettingsPage>
+</template>
+
+<route lang="yaml">
+meta:
+  admin: true
+</route>
+
+<script setup lang="ts">
+import {
+  BaseItemDtoQueryResult,
+  UnratedItem,
+  UserDto
+} from '@jellyfin/sdk/lib/generated-client';
+import { getUserApi } from '@jellyfin/sdk/lib/utils/api/user-api';
+import { onMounted, ref } from 'vue';
+import { useI18n } from 'vue-i18n';
+import { useRoute , useRouter } from 'vue-router';
+import { getLibraryApi } from '@jellyfin/sdk/lib/utils/api/library-api';
+import { getLocalizationApi } from '@jellyfin/sdk/lib/utils/api/localization-api';
+import { useConfirmDialog, useRemote } from '@/composables';
+
+const { t } = useI18n();
+const route = useRoute();
+const router = useRouter();
+const remote = useRemote();
+
+const loading = ref<boolean>(false);
+const addTagDialogOpen = ref<boolean>(false);
+const newTagValue = ref<string>('');
+const user = ref<UserDto>({});
+const libraries = ref<BaseItemDtoQueryResult>();
+const parentalCategories = ref<{ label: string, id: number | undefined }[]>([]);
+const initializedUser = ref<{ Name: string, CurrentPassword: string, Password: string, ConfirmPassword: string, CanAccessAllLibraries: boolean, Folders: string[], maxParentalRating: number | undefined, BlockUnratedItems: UnratedItem[], BlockedTags: string[] }>({Name: '', CurrentPassword: '', Password: '', ConfirmPassword: '', CanAccessAllLibraries: false, Folders: [], maxParentalRating: undefined, BlockUnratedItems: [], BlockedTags: []});
```

Comment:
Remove this and run everything inside the onMounted hook at the top level, so navigation is blocked until all the data is populated and you don't need this.

Reply:
Not quite sure what you mean here. The logic should already be inside the onMounted hook. I made `initializeUser()` a separate function since I need that in two different places.

Reply:
@rdmchr You don't need ``onMounted`` for anything here, just extract it's contents to another function, like this:

```
async load() {
...
}

await load()
```

Read about [Vue's 3 Suspense](https://vuejs.org/guide/built-ins/suspense.html#async-setup)

Reply:
Wow,, I didn't know it was that easy. Thanks for the tip 😄